lib/bundler/uri_credentials_filter.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+module Bundler
+  module URICredentialsFilter
+    def anonymized_uri(uri_to_anonymize)
+      return uri_to_anonymize if uri_to_anonymize.nil?
+      uri = uri_to_anonymize.dup
+      uri = URI(uri.to_s) unless uri.is_a?(URI)
+      uri.user = uri.password = nil if uri.userinfo
```

Comment:
On second thought, I think filtering out the password by default, and the user section (actually oauth token) instead of password when the password is 'x' or 'x-oauth-basic' covers all edge cases. I didn't consider the idea that the gem or git server or etc. wouldn't really allow regular login credentials that include 'x' or 'x-oauth-basic' as the password, because the server would try to authenticate using oauth in these cases. (i.e. `username123:x-oauth-basic@` would be authenticated as oauth where `username123` is the oauth token.)